daemon/bitcoin/helpers.go

```diff
@@ -0,0 +1,184 @@
+package bitcoin
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/ecdsa"
+	"github.com/btcsuite/btcd/btcutil/psbt"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/lntypes"
+	log "github.com/sirupsen/logrus"
+)
+
+func signInput(packet *psbt.Packet, inputIndex int, key *btcec.PrivateKey, sigHashType txscript.SigHashType, fetcher txscript.PrevOutputFetcher) ([]byte, error) {
+	if inputIndex < 0 || inputIndex >= len(packet.Inputs) {
+		return nil, fmt.Errorf("invalid input index: %d", inputIndex)
+	}
+
+	input := &packet.Inputs[inputIndex]
+
+	sigHashes := txscript.NewTxSigHashes(packet.UnsignedTx, fetcher)
+	sigHash, err := txscript.CalcWitnessSigHash(
+		input.WitnessScript,
+		sigHashes,
+		sigHashType,
+		packet.UnsignedTx,
+		inputIndex,
+		input.WitnessUtxo.Value,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	signature := ecdsa.Sign(key, sigHash)
+	sigWithHashType := append(signature.Serialize(), byte(sigHashType))
+
+	valid := signature.Verify(sigHash, key.PubKey())
+	if !valid {
+		return nil, fmt.Errorf("signature verification failed: %w", err)
+	}
```

Comment:
I dont think we need this anymore, this was mainly for debugging

Reply:
true